sonarcloud/helpers.go

```diff
@@ -100,3 +105,52 @@ func findProject(response *projects.SearchResponseAll, key string) (Project, boo
 	}
 	return result, ok
 }
+
+// findQualityGate returns the quality gate with the given name if it exists in a response
+func findQualityGate(response *qualitygates.ListResponse, name string) (QualityGate, bool) {
+	var result QualityGate
+	ok := false
+	for _, q := range response.Qualitygates {
+		if q.Name == name {
+			result = QualityGate{
+				ID:        types.String{Value: fmt.Sprintf("%d", int(q.Id))},
+				GateId:    types.Float64{Value: q.Id},
+				Name:      types.String{Value: q.Name},
+				IsBuiltIn: types.Bool{Value: q.IsBuiltIn},
+				IsDefault: types.Bool{Value: q.IsDefault},
+			}
+			for _, c := range q.Conditions {
+				result.Conditions = append(result.Conditions, Condition{
+					Error:  types.String{Value: c.Error},
+					ID:     types.Float64{Value: c.Id},
+					Metric: types.String{Value: c.Metric},
+					Op:     types.String{Value: c.Op},
+				})
+			}
+			ok = true
+			break
+		}
+	}
+	return result, ok
+}
+
+// findSelection returns a Selection{} struct with the given project keys if they exist in a response
+// this can be sped up using hashmaps, but I didn't feel like introducing a new dependency/taking code from somewhere.
+// Ex library: https://pkg.go.dev/github.com/juliangruber/go-intersect/v2
+func findSelection(response *qualitygates.SearchResponse, keys []attr.Value) (Selection, bool) {
+	projectKeys := make([]attr.Value, 0)
+	ok := true
+	for _, k := range keys {
+		ok = false
+		for _, s := range response.Results {
+			if k.Equal(types.String{Value: s.Key}) {
+				projectKeys = append(projectKeys, types.String{Value: strings.Trim(s.Key, "\"")})
+				ok = true
+				break
+			}
+		}
```

Comment:
Because we don't check for `ok` after each key, only the last iteration will determine the value of `ok`. What can happen is that the second-to-last project key is not present in the result, but because the last project key **is**, `ok` will be set to `true`, even though we are missing a project selection.

To fix this, we should check for `ok` being `false`, and break the outer loop if it is indeed `false`.

```suggestion
		}
		if !ok {
		  break
		}
```